pages/[...slug].tsx

```diff
@@ -1,58 +1,92 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import payload from 'payload';
 import { GetServerSideProps } from 'next';
-import getConfig from 'next/config';
+import { PayloadAdminBar, PayloadAdminBarProps } from 'payload-admin-bar';
+import { useRouter } from 'next/router';
 import { Type as PageType } from '../collections/Page';
 import NotFound from '../components/NotFound';
 import Head from '../components/Head';
 import classes from '../css/page.module.css';
 import RenderBlocks from '../components/RenderBlocks';
 
-const { publicRuntimeConfig: { SERVER_URL } } = getConfig();
-
 export type Props = {
   page?: PageType
   statusCode: number
+  pageID?: string
 }
 
-const Page: React.FC<Props> = (props) => {
-  const { page } = props;
+const appURL = process.env.NEXT_PUBLIC_SERVER_URL;
+
+const Page: React.FC<Props & PayloadAdminBarProps> = (props) => {
+  const { page, preview } = props;
+  const router = useRouter();
+
+  const onPreviewExit = useCallback(() => {
+    const exit = async () => {
+      const exitReq = await fetch('/api/exit-preview');
+      if (exitReq.status === 200) {
+        router.reload();
+      }
+    };
+    exit();
+  }, [router]);
 
   if (!page) {
     return <NotFound />;
   }
 
   return (
-    <main className={classes.page}>
+    <main>
       <Head
         title={page.meta?.title || page.title}
         description={page.meta?.description}
         keywords={page.meta?.keywords}
       />
-      <header className={classes.header}>
-        <h1>{page.title}</h1>
-      </header>
-      <div className={classes.featuredImage}>
-        {page.image && (
+      <PayloadAdminBar
```

Comment:
My only concern here would be using PayloadAdminBar in a real project, you really would need to abstract it out into the header component so it can easily be re-used. I am unsure if that is outside of the scope of this example though.